graph/GuidelinesGraph.md

```diff
@@ -0,0 +1,441 @@
+API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/graph/Guidelines.md#123-definition-of-a-breaking-change).
+In general, making all but additive changes to the API contract for existing elements is
+considered breaking. Adding new elements is allowed and not considered a
+breaking change.
+
+\*\* Non-breaking changes:\*\*
+
+
+-   Addition of properties that are nullable or have a default value
+-   Addition of a member to an evolvable enumeration 
+-   Removal, rename, or change to the type of an annotation   
+-   Changes to the order of properties
+-   Changes to the length or format of opaque strings, such as resource IDs
+-   Addition or removal of an annotation OpenType="true" 
+
+\*\* Breaking changes:\*\*
+
+-   Changes to the URL or fundamental request/response associated with a
+    resource
+-   Removal, rename, or change to an incompatible type of a declared property
+-   Removal or rename of APIs or API parameters
+-   Addition of a required request header
+-   Addition of a EnumType members for non-evolvable enumerations
+-   Addition of a Nullable="false" properties to existing types
+-   Addition of a Nullable="false" parameters to existing actions and functions
+-    Changes to top-level error codes
+-    Introduction of server-side pagination to existing collections
+-    Changes to the default order of collection elements
+-    Significant changes to the performance of APIs such as increased latency, rate limits or concurrency.
+
+
+
+### Versioning and Deprecation
+As the market and technology evolves your APIs will require modifications in this case you must avoid breaking changes and add new resources and features incrementally. If that is not possible then you must version elements of your APIs.
+Microsoft Graph allows versioning of elements including entities and properties. The versioning process goes along with deprecation and as soon as you introduce a new element update the previous version needs to follow the deprecation process.
+
+You must create a new version of your element for any breaking change and name it uniquely. 
+In some cases, the API will have evolved such that there is a new, natural unique name.   In other cases, the original name may still be the most descriptive for the evolved element.  In the latter case, the suffix _v2 must be added to the original name to make it unique.
+The original element is then marked as deprecated using annotations.
+
+Microsoft Graph provides two public endpoints to support API lifecycle:
+1.	API sets on the v1.0 endpoint (https://graph.microsoft.com/v1.0) are in general availability (GA) status.
+2.	API sets on the beta endpoint (https://graph.microsoft.com/beta) are in beta or private preview status.
+
+Microsoft Graph APIs in the GA version guarantee API stability and consistency for its clients. If your API requires a breaking change in GA, then you MUST create new element versions and support deprecated elements for a minimum of 36 months. 
```

Comment:
Technically, it's 36 months, or 24 months with demonstrated non-usage.

Reply:
agree